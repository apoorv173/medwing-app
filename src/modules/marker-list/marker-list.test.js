import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import MarkerList from './marker-list';

const positionArr = [{
    formatted_address: '',
    geometry: {
        location: {
            lat: () =>jest.fn(),
            lng: () =>jest.fn()
        }
    },
}];

it('Should render all the dom elements', () => {
	const onEdit = jest.fn();

    const onRemove = jest.fn();

	const markerList = shallow(<MarkerList
        positionArr={positionArr}
        onEditMarker={onEdit}
        onRemoveMarker={onRemove}
        formInEditMode={false}
    />
	);
    expect(markerList.find('h3').length).toEqual(1);
    expect(markerList.find('p').length).toEqual(3);
    expect(markerList.find('span').length).toEqual(4);
    expect(markerList.find('button').length).toEqual(2);
    expect(markerList.find('.content-wrapper').length).toEqual(1);
    expect(markerList.find('.col-xs-12').length).toEqual(1);

    markerList.setProps({formInEditMode: true});
    expect(markerList.find('button').length).toEqual(0);
});

it('should trigger the onEdit and onRemove method on button click', () => {
    const onEdit = jest.fn();

    const onRemove = jest.fn();

	const markerList = shallow(<MarkerList
        positionArr={positionArr}
        onEditMarker={onEdit}
        onRemoveMarker={onRemove}
        formInEditMode={false}
    />
    );
    markerList.find('.btn-primary').simulate('click');
    expect(onEdit).toHaveBeenCalled();

    markerList.find('.btn-secondary').simulate('click');
    expect(onRemove).toHaveBeenCalled();    
});

it('renders correctly', () => {
    const positionObj = {
        formatted_address: ''
    };
    const onEdit = jest.fn();

    const onRemove = jest.fn();

  const tree = renderer
    .create(<MarkerList
        positionArr={positionArr}
        onEditMarker={onEdit}
        onRemoveMarker={onRemove}
        formInEditMode={false}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});