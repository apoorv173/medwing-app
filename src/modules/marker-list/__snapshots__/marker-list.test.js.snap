// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="markers-list"
>
  <li
    className="col-xs-12 col-sm-6"
  >
    <div
      className="content-wrapper"
    >
      <div
        className="detail-section"
      >
        <h3>
          
        </h3>
        <hr />
        <p>
          
        </p>
        <p>
          <span>
            Latitude:
          </span>
          <span />
        </p>
        <p>
          <span>
            Longitude:
          </span>
          <span />
        </p>
      </div>
      <div
        className="btn-group"
      >
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          Edit
        </button>
        <button
          className="btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </li>
</ul>
`;
